File: js/0404.js
// ‰º~ŠKæ™p
function FallingFact(x, n) {
    if (n <= 1)
        return x
    else
        return x * FallingFact(x - 1, n - 1)
}

// 10 ‚©‚ç 6 ‚Ü‚Å‚Ì ‘æ
WScript.Echo(FallingFact(10, 5))
WScript.Echo(10 * 9 * 8 * 7 * 6)

File: pl/0404.pl
# ‰º~ŠKæ™p
sub FallingFact
{
    my ($x, $n) = @_;

    if ($n <= 1)
    {
        $x;
    }
    else
    {
        $x * FallingFact($x - 1, $n - 1);
    }
}

# 10 ‚©‚ç 6 ‚Ü‚Å‚Ì ‘æ
print FallingFact(10, 5), "\n";
print 10 * 9 * 8 * 7 * 6, "\n";
